src/main/scala/org/broadinstitute/dsde/workbench/sam/directory/JndiDirectoryDAO.scala

```diff
@@ -213,6 +231,18 @@ class JndiDirectoryDAO(protected val directoryConfig: DirectoryConfig)(implicit
     }
   }
 
+  override def loadSubjectEmail(subject: WorkbenchSubject): Future[Option[WorkbenchEmail]] = withContext { ctx =>
+    val subDn = subjectDn(subject)
+    Option(ctx.getAttributes(subDn).get(Attr.email)).map { emailAttr =>
+      val emailString = emailAttr.get.asInstanceOf[String]
+      subject match {
```

Comment:
Should we have a function on `WorkbenchSubject` that takes an email and returns an appropriate `WorkbenchEmail`?